### ISSUE TITLE
Change worker to process a block for mini-batch

### ISSUE THREAD

**Issue by wynot12** [snuspl/cruise#1165]
Currently, Dolphin's workers treat training data with key granularity. It's because ET only provides block-level access API.

However, it would be more natural to treat data in block-granularity and block size is same with the one of mini-batch. Then workers process a block for each mini-batch.

This way will greatly simplify metric-optimizer part by allowing them to deal with only the number of blocks, not number of actual data entries.
